File: ios/NativeReactIOS/MyEventEmitter.m
//
//  MyEventEmitter.m
//  NativeReactIOS
//
//  Created by Basit on 15/03/2023.
//
#import <Foundation/Foundation.h>
#import "React/RCTBridgeModule.h"
#import "React/RCTBridge.h"
#import "React/RCTEventDispatcher.h"
//@interface RCT_EXTERN_MODULE(MyEventEmitter, NSObject)
//RCT_EXTERN_METHOD(changeText:(NSString *)message)
//RCT_EXTERN_METHOD(sendMessageToReact:(NSString *)message)
//@end

@interface RCT_EXTERN_MODULE(MyEventEmitter, NSObject);

RCT_EXTERN_METHOD(emitEvent:(NSString *)body)
RCT_EXTERN_METHOD(receiveData:(NSArray *)body)

@end

File: ios/NativeReactIOS/MyEventEmitter.swift
import Foundation
//import React

@objc(MyEventEmitter)
class MyEventEmitter: RCTEventEmitter {
    public static var shared: MyEventEmitter?
    
    override init() {
        super.init()
        MyEventEmitter.shared = self
    }
    
  override func supportedEvents() -> [String]! {
    return ["myJavscriptFunction"]
  }

  @objc
  func emitEvent(_ body: String) {
     let appDelegate = UIApplication.shared.delegate as! AppDelegate //For IOS
//      let appDelegate = NSApplication.shared.delegate as! AppDelegate // For MAC
     appDelegate.bridge.eventDispatcher().sendAppEvent(withName: "myJavscriptFunction", body: body)
////      appDelegate.bridge.eventDispatcher().sendAppEvent(withName: "myJavscriptFunction", body: body)

//      appDelegate.bridge.eventDispatcher().sendAppEvent(withName: "myJavscriptFunction", body: body)
////      appDelegate.bridge.eventDispatcher().sendAppEvent(withName: "myJavscriptFunction", body: body)
      
//      sendEvent(withName: "myJavscriptFunction", body: "TEST SEND EVENT")

//      AppDelegate.shared.bridge.eventDispatcher().sendAppEvent(withName: "myJavscriptFunction", body: body)
//    sendEvent(withName: eventName, body: body)
  }
    
    @objc
    func receiveData(_ body: NSArray) {
        print("IN ON EVENT:")
        for item in body {
          let name = (item as! NSDictionary)["name"] as! String
          let isoCode = (item as! NSDictionary)["iso_code"] as! String
          print("Name: \(name), ISO Code: \(isoCode)")
        }
    }

  override static func requiresMainQueueSetup() -> Bool {
    return true
  }
}
